v-next/hardhat-ignition/src/internal/tasks/deploy.ts

```diff
@@ -0,0 +1,337 @@
+      executionEventListener.ledgerConfirmationSuccess();
+    const ledgerConfirmationFailure = () =>
+      executionEventListener.ledgerConfirmationFailure();
+
+    try {
+      await connection.provider.send("hardhat_setLedgerOutputEnabled", [false]);
+
+      connection.provider.once("connection_start", ledgerConnectionStart);
+      connection.provider.once("connection_success", ledgerConnectionSuccess);
+      connection.provider.once("connection_failure", ledgerConnectionFailure);
+      connection.provider.on("confirmation_start", ledgerConfirmationStart);
+      connection.provider.on("confirmation_success", ledgerConfirmationSuccess);
+      connection.provider.on("confirmation_failure", ledgerConfirmationFailure);
+    } catch (error) {
+      log(error);
+    }
+
+    const result = await deploy({
+      config: hre.config.ignition,
+      provider: connection.provider,
+      executionEventListener,
+      artifactResolver,
+      deploymentDir,
+      ignitionModule: userModule,
+      deploymentParameters: parameters ?? {},
+      accounts,
+      defaultSender: defaultSender === "" ? undefined : defaultSender,
+      strategy: strategyName,
+      strategyConfig,
+      maxFeePerGasLimit:
+        hre.config.networks[connection.networkName]?.ignition.maxFeePerGasLimit,
+      maxPriorityFeePerGas:
+        hre.config.networks[connection.networkName]?.ignition
+          .maxPriorityFeePerGas,
+      gasPrice: hre.config.networks[connection.networkName]?.ignition.gasPrice,
+      disableFeeBumping:
+        hre.config.ignition.disableFeeBumping ??
+        hre.config.networks[connection.networkName]?.ignition.disableFeeBumping,
+    });
+
+    try {
+      await connection.provider.send("hardhat_setLedgerOutputEnabled", [true]);
+
+      connection.provider.off("connection_start", ledgerConnectionStart);
+      connection.provider.off("connection_success", ledgerConnectionSuccess);
+      connection.provider.off("connection_failure", ledgerConnectionFailure);
+      connection.provider.off("confirmation_start", ledgerConfirmationStart);
+      connection.provider.off(
+        "confirmation_success",
+        ledgerConfirmationSuccess,
+      );
+      connection.provider.off(
+        "confirmation_failure",
+        ledgerConfirmationFailure,
+      );
+    } catch (error) {
+      log(error);
+    }
+
+    // TODO: HH3 Bring back with the port of hardhat-verify
+    // if (result.type === "SUCCESSFUL_DEPLOYMENT" && verify) {
+    //   console.log("");
+    //   console.log(chalk.bold("Verifying deployed contracts"));
+    //   console.log("");
+
+    //   await hre.run({ scope: "ignition", task: "verify" }, { deploymentId });
+    // }
+
+    if (result.type !== "SUCCESSFUL_DEPLOYMENT") {
+      process.exitCode = 1;
+    }
+
+    return result;
+  } catch (e) {
+    if (e instanceof IgnitionError && shouldBeHardhatPluginError(e)) {
+      throw new HardhatError(HardhatError.ERRORS.IGNITION.INTERNAL_ERROR, e);
```

Comment:
For the moment we are choosing to wrap `IgnitionError`'s with an internal `HardhatError`.
We intend to move IgnitionErrors into HardhatError proper during the Alpha.